tests/Zend/Form/ElementTest.php

```diff
@@ -1225,6 +1225,30 @@ public function testOptionsAreCastToArrayWhenAddingFilter()
         $this->assertTrue($filter instanceof Zend_Filter_Alnum);
         $this->assertTrue($filter->allowWhiteSpace);
     }
+    
+    public function testFilterOptions()
+    {
+        try {
+            $this->element->addFilter('Boolean', array('type' => Zend_Filter_Boolean::PHP + Zend_Filter_Boolean::FALSE_STRING));
+        } catch (Exception $e) {
+            $this->fail('Should be able to add array filter options');
+        }
+        $filter = $this->element->getFilter('Boolean');
+        $this->assertTrue($filter instanceof Zend_Filter_Boolean);
+        $this->assertSame($filter->getType(), Zend_Filter_Boolean::PHP + Zend_Filter_Boolean::FALSE_STRING);
+    }
+    
+    public function testFilterOptionsAsArray()
+    {
+        try {
+            $this->element->addFilter('HtmlEntities', array(array('encoding' => 'MacRoman')));
```

Comment:
I think this test case is redundant - nested assoc array in an array is already tested in the next test right below it - `testShouldUseFilterConstructorOptionsAsPassedToAddFilter()`

A test which checks if a filter can be instantiated with an array of arguments via a numerical array.

For instance, `addFilter('SeparatorToSeparator', array('-', '_'))`